sia-gpu-miner.c

```diff
@@ -302,10 +313,33 @@ void printPlatformsAndDevices() {
 			ret = clGetDeviceInfo(deviceids[j], CL_DEVICE_NAME, 80, str, NULL);
 			if (ret != CL_SUCCESS) {
 				printf("\tError while getting device info.\n");
-				free(deviceids);
-				continue;
-			}
-			printf("\tDevice %d: %s\n", j, str);
+            } else {
+                printf("\tDevice %d: %s\n", j, str);
+            }
```

Comment:
Also, if we can assume that all devices have names, then the rest of the info should be printed with a double tab instead of repeating the device id. For example:
```
    Device 0: MyDevice
        Endianness: Little-endian
        Address space: 64-bit
```
instead of:
```
    Device 0: MyDevice
    Device 0: Endianness: Little-endian
    Device 0: Address space: 64-bit
```
I suppose having a uniform prefix is handy if you want to pipe the output to `grep` or something, but I imagine you only need to do that if you have lots of devices.